connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -280,6 +298,25 @@ ServerConnector findConnector(String protocol) {
         return null;
     }
 
+    void registerRestExtensions(Herder herder, ResourceConfig resourceConfig) {
+        connectRestExtensions = config.getConfiguredInstances(
+            WorkerConfig.REST_EXTENSION_CLASSES_CONFIG,
+            ConnectRestExtension.class
+        );
```

Comment:
As I mentioned earlier, we need to use Java Service Provider API to find which plugin has the specified implementation, and we should change Plugins to support creating these instances. This code only works if the implementations are on the classpath.

Reply:
Yes Agreed. I still haven't integrated with the plugin classloader yet. Hence used this method for the draft so that  we get a better picture of the public interfaces.